packages/builder/src/schemas.zod.ts

```diff
@@ -0,0 +1,304 @@
+        abi: z.string(),
+        abiOf: z.array(z.string()),
+        args: z.array(z.any()),
+        libraries: z.record(z.string()),
+
+        // used to force new copy of a contract (not actually used)
+        salt: z.string(),
+
+        value: z.string(),
+        overrides: z.object({
+          gasLimit: z.string(),
+        }),
+
+        depends: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+export const importSchema = z
+  .object({
+    source: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        chainId: z.number().int(),
+        preset: z.string(),
+        depends: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+export const invokeSchema = z
+  .object({
+    target: z.array(z.string()).nonempty(),
+    func: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        abi: z.string(),
+
+        args: z.array(z.any()),
+        from: z.string(),
+        fromCall: z
+          .object({
+            func: z.string(),
+          })
+          .merge(
+            z.object({
+              args: z.array(z.any()),
+            })
+          ),
+        value: z.string(),
+        overrides: z.object({
+          gasLimit: z.string(),
+        }),
+        extra: z.record(
+          z.object({
+            event: z.string(),
+            arg: z.number().int(),
+
+            allowEmptyEvents: z.boolean().optional(),
+          })
+        ),
+        factory: z.record(
+          z.object({
+            event: z.string(),
+            arg: z.number().int(),
+
+            artifact: z.string().optional(),
+            abiOf: z.array(z.string()).optional(),
+            constructorArgs: z.array(z.any()).optional(),
+            allowEmptyEvents: z.boolean().optional(),
+          })
+        ),
+        depends: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+export const provisionSchema = z
+  .object({
+    source: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        chainId: z.number().int(),
+        sourcePreset: z.string(),
+        targetPreset: z.string(),
+        options: z.record(z.string()),
+        tags: z.array(z.string()),
+        depends: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+export const keeperSchema = z
+  .object({
+    exec: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        args: z.array(z.string()),
+        env: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+/**
+ * @internal NOTE: if you edit this schema, please also edit the constructor of ChainDefinition in 'definition.ts' to account for non-action components
+ */
+export const chainDefinitionSchema = z
+  .object({
+    name: z.string(),
+    version: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        description: z.string(),
+        keywords: z.array(z.string()),
+        setting: z.record(
+          z
+            .object({
+              description: z.string(),
+              type: z.enum(['number', 'string', 'boolean']),
```

Comment:
type probably isnt needed at this point, we havent ever used it tmk. do you think it still has a use?

Reply:
Havent really seen it used anywhere, so it's probably ok to remove. Will double check to make sure there isnt any package that happens to be using it so we avoid introducing breaking changes.